utils/general.py

```diff
@@ -36,6 +38,75 @@
 # Prevent OpenCV from multithreading (to use PyTorch DataLoader)
 cv2.setNumThreads(0)
 
+#: Download objects
+def download_file_blob_to_dir(directory, file_blob):
+    file_name = file_blob.name.split("/")[2]
+    destination = Path(directory) / f"{file_name}"
+    destination.touch()
+    blob.download_to_filename(destination)
+
+
+#: Copy GCS objects in their respective local directories
+def copy_object_from_GCS(local_directory, file_blob):
+    train_image_directory = local_directory[0]
+    train_label_directory = local_directory[1]
+    validation_image_directory = local_directory[2]
+    validation_label_directory = local_directory[3]
+
+    if file_blob.name.startswith("images/train") and file_blob.name.endswith(".jpg"):
+        download_file_blob_to_dir(train_image_directory, file_blobblob)
+
+    elif file_blob.name.startswith("labels/train") and file_blob.name.endswith(".txt"):
+        download_file_blob_to_dir(train_label_directory, file_blob)
+
+    elif file_blob.name.startswith("images/val") and file_blob.name.endswith(".jpg"):
+        download_file_blob_to_dir(validation_image_directory, file_blob)
+
+    elif file_blob.name.startswith("labels/val") and file_blob.name.endswith(".txt"):
+        download_file_blob_to_dir(validation_label_directory, file_blob)
+
+
+def download_training_objects_from_GC_bucket(func):
+    """
+    This decorator is called before the function check_dataset, which checks if a dataset is present
+    on the system. The function reads the .yaml file, which should consists of a google cloud storage bucket and a directory
+    where the training images are.
+    """
```

Comment:
Great! 😄 

Generally, documentation shouldn't dictate such specific usage as here. We can use this decorator on any function that takes a dictionary with an entry containing the name of a gc bucket.

Instead, explain why you have written the decorator.

Reply:
Fixed 1823a9594e94aa70ae86e3a80367c81d79b22c9e